### PR TITLE
Update smdebug to 0.7.0

### PR DESCRIPTION
**Pull request by vandanavk**
*Issue #, if available:*

*Description of changes:*
Using smdebug built from source. will be replaced with smdebug==0.7.0 once it is released.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by vandanavk** [comment]
CI is failing because of https://github.com/pytorch/vision/issues/1938

File: test/resources/mnist/smdebug_mnist.py [hunk 1]
```diff
@@ -30,6 +30,22 @@
...
+    class AppURLopener(urllib.FancyURLopener):
+        version = "Mozilla/5.0"
+
+    urllib._urlopener = AppURLopener()
+except AttributeError:
+    # For python 3
+    opener = urllib.request.build_opener()
+    opener.addheaders = [('User-agent', 'Mozilla/5.0')]
+    urllib.request.install_opener(opener)
+
```
Comment (codewithsk) [inline test/resources/mnist/smdebug_mnist.py]: Can you submit this patch as a separate PR and remove it from the current PR? It is required for the SMDebug tests to pass in other PRs as well?

**Comment by codewithsk** [comment]
LGTM. Please send another PR once SMDebug 0.7.0 is released on PyPI
